### PR TITLE
Feature/loose fk zaakbesluit

### PR DESCRIPTION
**Pull request by annashamray**
Issue - https://github.com/open-zaak/open-zaak/issues/296

### CODE REVIEW

File: src/openzaak/components/besluiten/models.py [hunk 1]
```diff
@@ -157,6 +157,29 @@ class Besluit(AuditTrailMixin, APIMixin, models.Model):
...
+        _("externe previous zaak"), blank=True, max_length=1000,
+    )
+    _previous_zaak = models.ForeignKey(
+        "zaken.Zaak",
+        on_delete=models.PROTECT,
+        null=True,
+        blank=True,  # een besluit kan niet bij een zaak horen
+        related_name="previous_zaak",
+    )
+    previous_zaak = FkOrURLField(
```
Comment (sergei-maertens) [inline src/openzaak/components/besluiten/models.py]: not sure why you're adding these as actual model fields?

File: src/openzaak/components/besluiten/models.py [hunk 2]
```diff
@@ -170,6 +193,17 @@ def __str__(self):
     def save(self, *args, **kwargs):
         if not self.identificatie:
             self.identificatie = generate_unique_identification(self, "datum")
+
+        # save previous zaak for triggers
+        if self.pk:
```
Comment (sergei-maertens) [inline src/openzaak/components/besluiten/models.py]: doesn't it make more sense to override `__init__` and store `self._previous_zaak = self.zaak`? You can optimize this by keeping the loose-fk underlying attributes and define a property

```py
@property
def previous_zaak(self):
    ...
```

to handle it, but I don't think keeping that information in sync in the DB will go well. I'm worried that will give all weird kind of problems to debug :thinking:

File: src/openzaak/components/zaken/models/zaken.py [hunk 3]
```diff
@@ -842,3 +848,57 @@ def save(self, *args, **kwargs):
...
+
+    class Meta:
+        verbose_name = _("zaakbesluit")
+        verbose_name_plural = _("zaakbesluiten")
+        unique_together = ("zaak", "_besluit")
+        constraints = [
+            models.UniqueConstraint(
+                fields=["zaak", "_besluit_url"],
+                condition=~models.Q(_besluit_url=""),
+                name="unique_zaak_and_besluit_document",
```
Comment (sergei-maertens) [inline src/openzaak/components/zaken/models/zaken.py]:
```suggestion
                name="unique_zaak_and_besluit",
```

File: src/openzaak/components/besluiten/models.py (hunk 1 above)
Comment (annashamray) [reply src/openzaak/components/besluiten/models.py]: `previous_zaak` fields are removed from db

File: src/openzaak/components/besluiten/models.py (hunk 2 above)
Comment (annashamray) [reply src/openzaak/components/besluiten/models.py]: the realization is changed to use `__init__` and `@property` instead of model fields

**Pull request merged**